### PR TITLE
[DOCS] Integrate feedback modal with Jira

### PR DESCRIPTION
**Pull request by JessSaavedra**
## Description
When a user fills the form in the feedback modal, a JIRA story is created in the documentation board

## Video
[Grabación de pantalla desde 17-07-24 13:16:02.webm](https://github.com/user-attachments/assets/ee95acb1-14f7-41ab-92d6-10ccd15c67be)

### Error message
image

- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [ ] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: docs/docusaurus/netlify/functions/createJiraTicketInDocsBoard.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+const formatOptionalValue = (value) => value || '-';
+
+const fullDescription = (description, name, email, selectedValue) => {
+    const formattedSelectedValue = selectedValue.replaceAll("-"," ");
+    return `Name: ${formatOptionalValue(name)}\nEmail: ${formatOptionalValue(email)}\nSelected feedback type: ${formattedSelectedValue}\n\n${description}`;
+}
+
+exports.handler = async (req, context) => {
+    const { description, name, email, selectedValue } = JSON.parse(req.body);
+    const response = await fetch("https://greatexpectations.atlassian.net/rest/api/2/issue", {
```
Comment (josectobar) [inline docs/docusaurus/netlify/functions/createJiraTicketInDocsBoard.js]: Nit, let's move the url to it's own variable for readability.

Comment (josectobar) [reply docs/docusaurus/netlify/functions/createJiraTicketInDocsBoard.js]: I think we should add a try catch to handle any exception.

File: docs/docusaurus/src/components/WasThisHelpful/index.js [hunk 2]
```diff
@@ -51,20 +52,24 @@ export default function WasThisHelpful(){
...
-                $survey_id: '018dd725-c595-0000-00c6-6eec1b197fd0',
-                $survey_response: formData.name,
-                $survey_response_1: formData.email,
-                $survey_response_2: formData.description,
-                $survey_response_3: pathname,
-                $survey_response_4: formData.selectedValue.replaceAll('-', ' ')
-            })
-            setIsOpen(false)
+        if (formData.description) {
+            const response = await fetch("/.netlify/functions/createJiraTicketInDocsBoard", {
```
Comment (josectobar) [inline docs/docusaurus/src/components/WasThisHelpful/index.js]: Same suggestion here, can we move this URL to a config file? we should also add a try-catch

File: docs/docusaurus/src/components/WasThisHelpful/index.js [hunk 3]
```diff
@@ -173,6 +178,8 @@ export default function WasThisHelpful(){
                         />
                     }
 
+                    {error && <p className={styles.errorMessage}>An error occured, please try again</p>}
```
Comment (josectobar) [inline docs/docusaurus/src/components/WasThisHelpful/index.js]:
```suggestion
                    {error && <p className={styles.errorMessage}>An error occurred, please try again later.</p>}
```

**Review by josectobar — CHANGES_REQUESTED** [review]
a couple of small suggestions, thank you!

File: docs/docusaurus/src/components/WasThisHelpful/index.js (hunk 2 above)
Comment (JessSaavedra) [reply docs/docusaurus/src/components/WasThisHelpful/index.js]: What do you mean by a config file, would you like me to add it somewhere else than a const? (i've changed that but i can take it to another file)

File: docs/docusaurus/netlify/functions/createJiraTicketInDocsBoard.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+const truncateDescription = (description) => {
+    return description.length > TITLE_MAX_CHARACTERS ? description.substring(0, TITLE_MAX_CHARACTERS) + "..." : description;
+}
+
+const formatOptionalValue = (value) => value || '-';
+
+const fullDescription = (description, name, email, selectedValue) => {
+    const formattedSelectedValue = selectedValue.replaceAll("-"," ");
+    return `Name: ${formatOptionalValue(name)}\nEmail: ${formatOptionalValue(email)}\nSelected feedback type: ${formattedSelectedValue}\n\n${description}`;
```
Comment (josectobar) [inline docs/docusaurus/netlify/functions/createJiraTicketInDocsBoard.js]: Can we add the bold format for the sections?

File: docs/docusaurus/netlify/functions/createJiraTicketInDocsBoard.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+exports.handler = async (req, context) => {
+    const { description, name, email, selectedValue } = JSON.parse(req.body);
+    try {
+        const response = await fetch(CREATE_JIRA_ISSUE_ENDPOINT_URL, {
+            method: "POST",
+            headers: {
+                'Content-Type': 'application/json',
+                'Authorization': 'Basic ' + Buffer.from(process.env.JIRA_API_USER + ":" + process.env.JIRA_API_TOKEN).toString('base64')
```
Comment (josectobar) [inline docs/docusaurus/netlify/functions/createJiraTicketInDocsBoard.js:24]: Let's wait until we have this service user situation buttoned up before we merge this.

**Review by josectobar — CHANGES_REQUESTED** [review]
A few suggestions. overall we should be good to merge but we need to address the JIRA service user situation first since we might need to use another alternative API.

File: docs/docusaurus/src/components/WasThisHelpful/index.js (hunk 2 above)
Comment (josectobar) [reply docs/docusaurus/src/components/WasThisHelpful/index.js]: Yup, you got it, I think moving it to a variable should work for now.

File: docs/docusaurus/src/components/WasThisHelpful/index.js [hunk 6]
```diff
@@ -51,20 +53,29 @@ export default function WasThisHelpful(){
...
-            posthog.capture("survey sent", {
-                $survey_id: '018dd725-c595-0000-00c6-6eec1b197fd0',
-                $survey_response: formData.name,
-                $survey_response_1: formData.email,
-                $survey_response_2: formData.description,
-                $survey_response_3: pathname,
-                $survey_response_4: formData.selectedValue.replaceAll('-', ' ')
-            })
-            setIsOpen(false)
+        if (formData.description) {
```
Comment (josectobar) [inline docs/docusaurus/src/components/WasThisHelpful/index.js:61]: Should we set error to false before we hit the API if it was previously marked as `true`?

**Review by josectobar — APPROVED** [review]
Just a small last change but everything else looks good.

**Pull request merged**